environs/bootstrap/config.go

```diff
@@ -194,15 +204,23 @@ func (c Config) Validate() error {
 	if len(c.ControllerExternalIPs) > 1 && c.ControllerServiceType == string(caas.ServiceLoadBalancer) {
 		return errors.NewNotValid(nil, fmt.Sprintf("only 1 external IP is allowed with service type %q", caas.ServiceLoadBalancer))
 	}
+	if c.SSHServerHostKey != "" {
+		// The errors given by this can be kind of misleading, if the key is not
+		// of the correct format, it will say "no key found", which is not very
+		// helpful. What it really means is "no valid key found".
+		if _, err := ssh.ParsePrivateKey([]byte(c.SSHServerHostKey)); err != nil {
+			return errors.Annotatef(err, "validating %s", SSHServerHostKeyKey)
+		}
+	}
 	return nil
 }
 
 // NewConfig creates a new Config from the supplied attributes.
 // Default values will be used where defaults are available.
 //
-// If ca-cert or ca-private-key are not set, then we will check
-// if ca-cert-path or ca-private-key-path are set, and read the
-// contents. If none of those are set, we will look for files
+// If ca-cert, ca-private-key or ssh-server-host-key are not set,
+// then we will check if ca-cert-path, ca-private-key-path ssh-server-host-key,
```

Comment:
To match the others?
```suggestion
// then we will check if ca-cert-path, ca-private-key-path ssh-server-host-key-path,
```

Reply:
Done